### PR TITLE
fix(predictions): add equatable conformance to errors

### PR DESCRIPTION
**Pull request by atierian**
## Description

- adds `Equatable` conformance to errors.
- add some error mapping test cases

## General Checklist

- [x] Added new tests to cover change, if needed
- [x] Build succeeds with all target using Swift Package Manager
- [x] All unit tests pass
- [x] All integration tests pass
- [x] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)
- [ ] ~Documentation update for the change if required~
- [x] PR title conforms to conventional commit style
- [ ] ~New or updated tests include `Given When Then` inline code documentation and are named accordingly `testThing_condition_expectation()`~
- [ ] ~If breaking change, documentation/changelog update with migration instructions~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Amplify/Categories/Predictions/Error/PredictionsError+ClientError.swift [hunk 1]
```diff
@@ -8,7 +8,12 @@
 import Foundation
 
 extension PredictionsError {
-    public struct ClientError {
+    public struct ClientError: Equatable {
+        public static func == (lhs: PredictionsError.ClientError, rhs: PredictionsError.ClientError) -> Bool {
+            lhs.description == rhs.description
```
Comment (royjit) [inline Amplify/Categories/Predictions/Error/PredictionsError+ClientError.swift:13]: Does comparing the descriptions the right way to see if the errors are equal?
Shouldnt we compare the error type instead?

Comment (atierian) [reply Amplify/Categories/Predictions/Error/PredictionsError+ClientError.swift:13]: We can, but the `underlyingError` is optional, so it won't always be there.

**Review by royjit — APPROVED** [review]

**Comment by royjit** [comment]
Unit test is failing

**Comment by atierian** [comment]
> Unit test is failing

Yup, the previous PR #2912 fixes that. They should pass on merging.

**Pull request merged**